Automattic/wp-calypso
# Hosting Command Palette: Display all the commands in scroll in Calypso and move up by context

Closes: https://github.com/Automattic/dotcom-forge/issues/4673

## Proposed Changes

This PR does the following:

* it makes the command palette display the list of all commands by default (with a side scroll)
* if the current path e.g. `/hosting-config` matches the command, it will rank higher. This means if the context of the commands match the current path, these commands will be pushed to the top
* it ranks the command `View my sites` on the top when you are not on the `/sites` pages and hides the `View my sites` from the `/sites` page

## Testing Instructions

**To test the display of all commands**

* Navigate to Calypso Live Link
* Navigate to `/sites`
* Press `cmd + k` on macOS or `ctrl + k` on Windows to trigger the command palette
* Observe that they can see the list of all commands with the scroll on the side

<img width="1262" alt="Screenshot 2023-12-01 at 11 07 37 AM" src="https://github.com/Automattic/wp-calypso/assets/25575134/0625dd41-b7d9-4f9a-82e5-c2aed83845ad">

* Select `Open site dashboard` command and select a site to navigate to
* Once in the site dashboard, press `cmd + k` on macOS or `ctrl + k` on Windows to trigger the command palette
* Confirm that you can still see the list of all the commands available

**To test the contextual commands ranking higher and View my sites command ranking first**:

* Navigate to Calypso Live Link
* Navigate to `/sites`
* Press `cmd + k` on macOS or `ctrl + k` on Windows to trigger the command palette
* Select `Manage hosting configuration` command 
* Select a site to navigate to
* Once in the Hosting Configuration on the site, press `cmd + k` on macOS or `ctrl + k` on Windows to trigger the command palette
* Observe that `View my sites` command ranking first:

<img width="960" alt="Screenshot 2023-12-01 at 11 19 45 AM" src="https://github.com/Automattic/wp-calypso/assets/25575134/221bf894-ddaa-4a03-883d-67e1a7c20c69">

* Observe that the second commands displayed is `Manage staging sites` - this is the command that has the context set to `hosting-config`:

<img width="1272" alt="Screenshot 2023-12-01 at 11 21 24 AM" src="https://github.com/Automattic/wp-calypso/assets/25575134/5a103c9c-5500-452d-93d8-e19b10533f31">

* Observe that the rest of the commands come in the order that they are added
* Navigate to `View my sites` command
* Press `cmd + k` on macOS or `ctrl + k` on Windows to trigger the command palette
* Confirm that `View my sites` command is not visible on `/sites` page

<img width="1208" alt="Screenshot 2023-12-01 at 11 34 34 AM" src="https://github.com/Automattic/wp-calypso/assets/25575134/5e6d3644-081c-43bd-a8a2-efce36a52ae2">

**Please note**: I only added the logic for the contextual commands to rank higher and so far added the context only to one command since I don't have a list of which context we want to assign to which command. Perhaps we can tweak the context for each command in the follow-up PR? cc @sejas 

## Pre-merge Checklist
